File: tree-editor-frontend/local_node_modules/treedocumentmodel/ActionConfiguration.ts
import {Action} from "./Action";
import {OrderedSet} from "crossecore/lib/OrderedSet";
import {Set} from "crossecore/lib/Set";
import {EMap} from "crossecore/lib/EMap";
import {InternalEObject} from "crossecore/lib/InternalEObject";

export interface ActionConfiguration
extends InternalEObject

{
	
	availableActions: OrderedSet<Action>;
	defaultActionbarActions: OrderedSet<Action>;
	nodeActions:EMap<string, string>;
	

}

File: model-server/src/main/java/no/ntnu/stud/krirek/treelsp/model/tree/ActionConfiguration.java
package no.ntnu.stud.krirek.treelsp.model.tree;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;
import java.util.Map;

@Value.Immutable
public interface ActionConfiguration {

    @NotNull List<Action> availableActions();

    /**
     * {@code List<ActionId>}
     */
    @Nullable List<String> defaultAcionbarActions();

    /**
     * {@code Map<ActionId, NodeType>}
     */
    @Nullable Map<String, @NotNull List<String>> nodeActions();
}
